uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/WorkDay.php

```diff
@@ -165,27 +157,55 @@ private static function validateWorkHours($params) {
       );
     }
 
-    $hasTimesAndBreak = $timeFrom && $timeTo && $break;
-    if($isWorkingDay && !$hasTimesAndBreak) {
+    $hasAllRequiredFields = !is_null($timeFrom) && !is_null($timeTo) && !is_null($break) && !is_null($numberOfHours);
+    if($isWorkingDay && !$hasAllRequiredFields) {
       throw new InvalidWorkDayException(
-        'Time From, Time To and Break are required for Working Days'
+        'Time From, Time To, Break and Number of Hours are required for Working Days'
       );
     }
 
     $hasTimes = !is_null($timeFrom) && !is_null($timeTo);
     if($hasTimes && (strtotime($timeFrom) >= strtotime($timeTo))) {
       throw new InvalidWorkDayException(
-          'Time From should be less than Time To'
+        'Time From should be less than Time To'
       );
     }
 
+    if (!is_null($break)) {
+      if (!is_numeric($break)) {
+        throw new InvalidWorkDayException(
+          'Break should be a valid number'
+        );
+      } else if ($break < 0) {
```

Comment:
No need for else if here, the condition can be in its own independent if.

Reply:
True that, it will stop once thrown anyway.

Reply:
Done